### ISSUE TITLE
`indent`: allow report error when the indentation of the comment is the same as the previous line

### ISSUE THREAD

**Issue by zzxming** [eslint-stylistic/eslint-stylistic#966]
### Clear and concise description of the problem

If a comment matches the expected indentation of the token immediately before or after, will not report the error. And this situation feels strange to me.

```ts
const obj = {
// this
  a: {
  // this
  },
  b: [
  // this
  ],
}
```

For me, the comment should have one more level indentation.

### Suggested solution

Add an option to control the `continue` branch.

https://github.com/eslint-stylistic/eslint-stylistic/blob/c58673516abbd28166a296631b638bd2198b34cd/packages/eslint-plugin/rules/indent/indent.ts#L2304-L2309

But now there are already a lot of options for 'indent', I don't know if this is a suitable request or solution. 

### Additional context

https://eslint-online-playground.netlify.app/#eNqdVMtOwzAQ/JWVL5GqJuEhJAhC4sAdJI51kUK6KS6pY9kOCqr676wfKWlRefVQRTs7s+PxJhtmdJULucA+s4YVrGqlsdA+r+AGNlzmOdgXYbgEKAtXABiVtlP3/1zA7ACYE7BlU4amEdJmJFqLZbZyA8RatdqCNQ+lNqih1u0aklv7rtBUWiibBlKuPJ5wGRmq6ZZCPtr3RhgrqoFohkIeeGnoIx6X2HvmAuuya2xw6c8AUc3EM7nfSCopDqeFpnBeAN01eITqs5SWFGYJat3qZApn80gfMotpxQQM2k6NrNXCq8+SySSf0K0kLk0HNKVcduUS75UVdE0jAyGrYpfqwbyj0v0/tCNhAL4SAFamp37d4VDyTn42tBrO+m3P703/3RuX82suaXNVWb2SMo1rJW2tF+BMlmvkrKCnuGy7e081Kt1y5p1x5tY5NK7bBW3LDlBavJXWY85ErIbNN64arXLm9MezIIsiA1jUot9vgDT1tc8IOVvg2x0qdFtZCdwfcezlCaoN+TRkYTf06Esa+p8us/PT7GRECE0RvcrOLrLTaI4+DxTz9gNR4m38

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by 9romise** [comment]
This is indeed a strange edge case – it reports errors for excessive indentation, but we also seem to have difficulty defining the correct indentation for comments.

**Comment by zzxming** [comment]
My previous idea may have been wrong. Indeed, There are many structures that have the same problem not only bracket. 

```ts
const result = func()
// this
  .foo()
  .bar()

const isTrue = true 
// this
  ? 'a' 
  : 'b'

function func() {
// this
}
```
I haven't looking for the detail yet. Except for the comments at the end of the block, in other cases, the comment should have same indentation with next token? Maybe special handling is needed when the next line is the closing bracket.

**Comment by MulverineX** [comment]
Related:

<img width="1109" height="311" alt="Image" src="https://github.com/user-attachments/assets/fa752116-ded7-4a74-b13d-dd1748a6dde5" />
